### ISSUE TITLE
Redesign messaging 

### ISSUE THREAD

**Issue by jbeirer** [fcs-proj/FastCaloSim#49]
To comply with messaging in Athena, our messaging currently relies on preprocessor macros. As those are identical with what is used in some Athena classes, we'll rename the ATH_MSG to FCS_MSG for now. Eventually, we should re-design the messaging to not use macros at all - see also https://github.com/fcs-proj/FastCaloSim/pull/48

**Comment by TheRealRobinHood** [comment]
Would you like to retain the logging architecture? So that inheritance is used or would you like to have the logging class detached, e.g. through a singleton or dependency injection?

**Comment by jbeirer** [comment]
I fear that to remain consistent with our input files e.g. https://github.com/fcs-proj/FastCaloSim/tree/main/test/data/param/barrel we will need to keep the logging structure as is for now. The plan is to port the code responsible for generating the input files to this repo soon, but this is still work in progress

**Comment by TheRealRobinHood** [comment]
Thank you very much for your feedback. I am currently working on the feature, but it may take some time as I don't have time every day after work.

**Comment by TheRealRobinHood** [comment]
I am afraid that it is not possible to improve the structure without regenerating the input files. This is because changing the MLogging class also changes the binary structure of the class. I have rewritten the MLogging class so that preprocessor statements are no longer necessary. So far, however, I have only implemented the class in the header file. The header and source file still need to be split up.

https://github.com/TheRealRobinHood/FastCaloSim/redesign-messaging

I suggest waiting until the repository responsible for generating the input files has been ported to this repository before finally re-implementing the MLogging class.

I believe this approach would provide the most performant solution.

**Comment by jbeirer** [comment]
Yes, agreed. There are some more urgent things on my list before I can focus on porting that part of the code, but we will try to get this in asap. In any case, thanks for this already!
